### PR TITLE
Add get_bounds method

### PR DESCRIPTION
**Pull request by marpulli**
Add get_bounds method to parameter space instead of having to convert to gpyopt design space.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: emukit/core/discrete_parameter.py [hunk 1]
```diff
@@ -32,6 +32,13 @@ def check_in_domain(self, x: Union[Iterable, float]) -> bool:
             x = [x]
         return set(x).issubset(set(self.domain))
 
+    @property
+    def bounds(self) -> List[Tuple]:
+        """
+        Returns tuple of min and max values parameter can take
```
Comment (apaleyes) [inline emukit/core/discrete_parameter.py]:
```suggestion
        Returns a list containing one tuple of min and max values parameter can take
```

File: emukit/core/parameter.py [hunk 2]
```diff
@@ -21,6 +21,13 @@ def model_parameters(self) -> List:
         """
         return [self]
 
+    @property
+    def bounds(self) -> List[Tuple]:
+        """
+        Returns the minimum and maximum values for the encoded version of the parameter
```
Comment (apaleyes) [inline emukit/core/parameter.py]: I'd suggest just "Returns bounds of the parameter in a form of list of tuples" or something like that. Mentioning encodings here might confuse people, as there is no encoding taking place for most parameters

File: emukit/core/parameter.py [hunk 3]
```diff
@@ -21,6 +21,13 @@ def model_parameters(self) -> List:
...
+        """
+        raise NotImplemented()
```
Comment (apaleyes) [inline emukit/core/parameter.py]:
```suggestion
        raise NotImplemented
```

File: tests/emukit/core/test_parameter_space.py [hunk 4]
```diff
@@ -49,3 +50,11 @@ def test_duplicate_parameter_names_fail():
...
     with pytest.raises(ValueError):
         ParameterSpace([p1, p2])
+
+
+def test_get_bounds():
+    p1 = ContinuousParameter('c', 1.0, 5.0)
+    p2 = DiscreteParameter('d', [1, 2, 3])
+    p3 = CategoricalParameter('cat', OneHotEncoding(['red', 'blue']))
+    space = ParameterSpace([p1, p2, p3])
+    assert space.get_bounds() == [(1., 5.), (1., 3.), (0, 1), (0, 1)]
```
Comment (apaleyes) [inline tests/emukit/core/test_parameter_space.py]: Flake won't like "no new line" thing

File: tests/emukit/core/test_parameter_space.py [hunk 5]
```diff
@@ -49,3 +50,11 @@ def test_duplicate_parameter_names_fail():
 
     with pytest.raises(ValueError):
         ParameterSpace([p1, p2])
+
+
+def test_get_bounds():
+    p1 = ContinuousParameter('c', 1.0, 5.0)
+    p2 = DiscreteParameter('d', [1, 2, 3])
+    p3 = CategoricalParameter('cat', OneHotEncoding(['red', 'blue']))
```
Comment (apaleyes) [inline tests/emukit/core/test_parameter_space.py]: Red and blue cats? Huh!

Comment (marpulli) [reply tests/emukit/core/test_parameter_space.py]: Changed to valid cat breeds

File: tests/emukit/core/test_parameter_space.py (hunk 4 above)
Comment (marpulli) [reply tests/emukit/core/test_parameter_space.py]: Added line

File: emukit/core/parameter.py (hunk 2 above)
Comment (marpulli) [reply emukit/core/parameter.py]: Changed

File: tests/emukit/core/test_parameter_space.py (hunk 5 above)
Comment (apaleyes) [reply tests/emukit/core/test_parameter_space.py]: Impossible to not approve after that

**Pull request merged**